src/web/src/components/Posts/Timeline.tsx

```diff
@@ -48,10 +49,25 @@ const Timeline = ({ pages, nextPage }: Props) => {
 
   // Iterate over all the pages (an array of arrays) and then convert all post
   // elements to <Post>
-  const postsTimeline = pages.map((page: Post[]): Array<ReactElement> | ReactElement =>
-    page.map(({ id, url }: Post) => <PostComponent postUrl={url} key={id} />)
-  );
 
+  const postsTimeline: ReactElement[] = [];
+  {
+    let previousCount = pages[0].length;
+
+    for (let i = 0; i < pages.length; i += 1) {
+      for (let j = 0; j < pages[i].length; j += 1) {
+        postsTimeline.push(
+          <PostComponent
+            postUrl={pages[i][j].url}
+            key={pages[i][j].id}
+            currentPost={i * previousCount + j + 1}
+            totalPosts={totalPosts}
+          />
+        );
+      }
+      previousCount = pages[i].length;
+    }
+  }
```

Comment:
This is a good idea. Let's achieve the same result and get rid of the key prop warning using `map`.
```
  const postsTimeline = pages.map((page, pageIndex): Array<ReactElement> | ReactElement =>
    page.map(({ id, url }, postIndex) => (
      <PostComponent
        postUrl={url}
        key={id}
        currentPost={pageIndex * pages[0].length + postIndex + 1}
        totalPosts={totalPosts}
      />
    ))
  );

```

Reply:
Yeah this is a good idea but I'm wondering if the amount of returned posts can be different everytime. Like the first can be 5 and then 4 and then 5. Something likes this. If not I can use the solution you recommend.

Reply:
I don't think so. The number of posts per page is a constant. The last page could have less than `n` posts though, but that doesn't affect the algorithm above.

Reply:
Turn `pageIndex * pages[0].length + postIndex + 1` into a function so it has a name, and is better documented.  Also, if you need to access `pages[0].length` over and over again, cache it in a variable above.